pandas/tests/reshape/merge/test_merge.py

```diff
@@ -2571,3 +2571,18 @@ def test_mergeerror_on_left_index_mismatched_dtypes():
     df_2 = DataFrame(data=["X"], columns=["C"], index=[999])
     with pytest.raises(MergeError, match="Can only pass argument"):
         merge(df_1, df_2, on=["C"], left_index=True)
+
+
+def test_merge_outer_with_NaN():
```

Comment:
can you parameterize with Int64 as well

Reply:
Can we switch left and right too? Changing lvals to rvals above would have fixed the issue but caused the inverse to fail

Reply:
done and done